locust_tasks/locustfile.py

```diff
@@ -166,49 +189,166 @@ def correct_address(self):
+            verify_response('RequestUacPost-Start', self, response, 200, Page.START)
+
+    @task
+    def enter_postcode(self):
+        """
+        POST postcode
+        """
+        with self.client.post("/en/requests/access-code/enter-address/", {
+            'form-enter-address-postcode': self.case['postcode']
+        }, catch_response=True) as response:
+            self.address_to_select = extractAddress(response, self.case["uprn"])
+            verify_response('RequestUacPost-EnterAddress', self, response, 200, Page.SELECT_ADDRESS,
+                            self.case["postcode"])
+
+    @task
+    def select_address(self):
+        """
+        POST uprn and whole address extracted as JSON from the HTML in previous task
+        """
+        with self.client.post("/en/requests/access-code/select-address/", {
+            'form-select-address': self.address_to_select
+        }, catch_response=True) as response:
+            verify_response('RequestUacPost-SelectAddress', self, response, 200, Page.ADDRESS_CORRECT, self.case["postcode"])
+
+    @task
+    def confirm_address(self):
+        """
+        POST 'yes' to confirm address
+        """
+        with self.client.post("/en/requests/access-code/confirm-address/", {
+            'form-confirm-address': 'yes'
+        }, catch_response=True) as response:
+            verify_response('RequestUacPost-ConfirmAddress', self, response, 200, Page.SELECT_METHOD, "Post")
+
+    @task
+    def select_method(self):
+        """
+        POST 'post' to select post as method of sending UACs
+        """
+        with self.client.post("/en/requests/access-code/select-method/", {
+            'form-select-method': 'post'
+        }, catch_response=True) as response:
+            verify_response('RequestUacPost-SelectMethod', self, response, 200, Page.ENTER_NAME)
+
+    @task
+    def enter_name(self):
+        """
+        POST 'John' as first name and 'Smith' as last name of person to send the UAC to
+        """
+        self.whole_name = self.case["first_name"] + " " + self.case["last_name"]
+        #logger.info("Name: " + self.whole_name)
+        with self.client.post("/en/requests/access-code/enter-name/", {
+            'name_first_name': self.case["first_name"],
+            'name_last_name': self.case["last_name"]
+        }, catch_response=True) as response:
+            verify_response('RequestUacPost-EnterName', self, response, 200, Page.CONFIRM_NAME,
+                            self.whole_name + "<br>")
+
+    @task
+    def confirm_name_address(self):
+        """
+        POST 'yes' to confirm name and address
+        """
+        with self.client.post("/en/requests/access-code/confirm-name-address/", {
+            'request-name-address-confirmation': 'yes'
+        }, catch_response=True) as response:
+            verify_response('RequestUacPost-ConfirmName', self, response, 200, Page.CODE_SENT, self.whole_name)
```

Comment:
As with previous step I think it would be clearer if the expected name is created just before it's needed.

Reply:
ok

Reply:
It gets created in the previous method, where it is initially used, so it doesn't need to be created here too does it?